### PR TITLE
Add getVoucherProxyCodeHash(..) & isRefundedTask(..) view functions

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: contracts/beacon/IVoucher.sol [hunk 1]
```diff
@@ -38,4 +38,5 @@ interface IVoucher {
     function getBalance() external view returns (uint256);
     function isAccountAuthorized(address account) external view returns (bool);
     function getSponsoredAmount(bytes32 dealId) external view returns (uint256);
+    function isRefundedTask(bytes32 taskId) external view returns (bool);
```
Comment (gfournieriExec) [inline contracts/beacon/IVoucher.sol:41]: switch places with `getSponsoredAmount`

Comment (jeremyjams) [reply contracts/beacon/IVoucher.sol:41]: Not sure to understand why (?):
```
    struct VoucherStorage {
        address _owner;
        address _voucherHub;
        uint256 _expiration;
        uint256 _type;
        mapping(address => bool) _authorizedAccounts;
        mapping(bytes32 dealId => uint256) _sponsoredAmounts;
        // Save refunded tasks to disable replay attacks.
        mapping(bytes32 taskId => bool) _refundedTasks;
    }
```

Comment (jeremyjams) [reply contracts/beacon/IVoucher.sol:41]: We try to order functions by visibility first, then by storage order (for getter), wdt?

Comment (gfournieriExec) [reply contracts/beacon/IVoucher.sol:41]: For the order: Yes mb thank you for highlighting this point

File: scripts/voucherHubUtils.ts [hunk 2]
```diff
@@ -39,58 +40,56 @@ export async function upgradeProxy(
...
+    } else {
+        /**
+         * @dev This is the voucher proxy code hash value expected in production,
+         * where contracts are deployed through a generic factory, hence having
+         * deterministic addresses.
+         *
+         * Note: Look very carefully before updating this value to avoid messing with
+         * existing vouchers already deployed in production.
+         */
+        return '0xebe0c40d5e931d865b61665b4e9435cc65ece10955457af1b4271e3438e4949e';
```
Comment (gfournieriExec) [inline scripts/voucherHubUtils.ts:93]: How did you compute this

**Review by gfournieriExec — APPROVED** [review]
Thanks

File: scripts/voucherHubUtils.ts (hunk 2 above)
Comment (jeremyjams) [reply scripts/voucherHubUtils.ts:93]: It's actually just the value we get at this time..! (but we could compute it with the formula above)

Comment (jeremyjams) [reply scripts/voucherHubUtils.ts:93]: The original need was to guaranty an hardcoded hash value that should not change on future upgrades, but we need to handle other cases too (no factory, coverage, ..)

File: scripts/voucherHubUtils.ts [hunk 3]
```diff
@@ -39,58 +40,56 @@ export async function upgradeProxy(
     const voucherHubUpgrade = contractUpgrade as VoucherHub;
     await voucherHubUpgrade.waitForDeployment();
     const voucherBeaconAddress = await voucherHubUpgrade.getVoucherBeacon();
-    const expectedHashes = await getVoucherProxyCreationCodeHash(voucherBeaconAddress);
-    const actualHash = await getVoucherProxyCreationCodeHashFromStorage(voucherHubAddress);
-    if (!expectedHashes.includes(actualHash)) {
+    const expectedHash = await getExpectedVoucherProxyCodeHash(voucherBeaconAddress);
+    const actualHash = await voucherHubUpgrade.getVoucherProxyCodeHash();
+    if (actualHash != expectedHash) {
```
Comment (zguesmi) [inline scripts/voucherHubUtils.ts:45]:
```suggestion
    if (actualHash !== expectedHash) {
```

**Review by zguesmi — APPROVED** [review]

**Pull request merged**